lang/cs/Org.Apache.REEF.Examples.HelloREEF/HelloDriver.cs

```diff
@@ -0,0 +1,120 @@
+{
+    /// <summary>
+    ///     The Driver for HelloREEF: It requests a single Evaluator and then submits the HelloTask to it.
+    /// </summary>
+    public sealed class HelloDriver : IObserver<IAllocatedEvaluator>, IObserver<IEvaluatorRequestor>, IStartHandler
+    {
+        /// <summary>
+        ///     Contexts contain configuration data used beyond a single task.
+        /// </summary>
+        private static readonly IConfiguration ContextConfiguration =
+            Driver.Context.ContextConfiguration.ConfigurationModule
+                .Set(Driver.Context.ContextConfiguration.Identifier, "HelloContext")
+                .Build();
+
+        /// <summary>
+        ///     The TaskConfiguration contains the type of Task to run as well as the identifier of that task
+        /// </summary>
+        private static readonly IConfiguration TaskConfiguration = Common.Tasks.TaskConfiguration.ConfigurationModule
+            .Set(Common.Tasks.TaskConfiguration.Identifier, "HelloTask")
+            .Set(Common.Tasks.TaskConfiguration.Task, GenericType<HelloTask>.Class)
+            .Build();
+
+        private readonly REEFFileNames _fileNames;
+
+        [Inject]
+        private HelloDriver(REEFFileNames fileNames)
+        {
+            _fileNames = fileNames;
+            ClrHandlerHelper.GenerateClassHierarchy(GetGlobalAssemblies());
+        }
+
+        /// <summary>
+        ///     Submits the HelloTask to the Evaluator.
+        /// </summary>
+        /// <param name="allocatedEvaluator"></param>
+        public void OnNext(IAllocatedEvaluator allocatedEvaluator)
+        {
+            allocatedEvaluator.SubmitContextAndTask(ContextConfiguration, TaskConfiguration);
+        }
+
+        public void OnError(Exception error)
+        {
+            throw error;
+        }
+
+        public void OnCompleted()
+        {
+        }
+
+        /// <summary>
+        ///     Ask for one Evaluator with 64MB of memory.
+        /// </summary>
+        /// <param name="evaluatorRequestor"></param>
+        public void OnNext(IEvaluatorRequestor evaluatorRequestor)
+        {
+            evaluatorRequestor.Submit(new EvaluatorRequest(1, 64));
+        }
+
+        public string Identifier { get; set; }
+
+        /// <summary>
+        /// </summary>
+        /// <returns>All DLLs in the global folder</returns>
+        private ISet<string> GetGlobalAssemblies()
+        {
+            return new HashSet<string>(Directory.GetFiles(_fileNames.GetGlobalFolderPath())
+                .Where(e => !(string.IsNullOrWhiteSpace(e)))
+                .Select(Path.GetFullPath)
+                .Where(File.Exists)
+                .Where(IsAssembly)
+                .Select(Path.GetFileNameWithoutExtension));
+        }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="path"></param>
+        /// <returns>True, if the path refers to an EXE or DLL</returns>
+        private static Boolean IsAssembly(string path)
```

Comment:
Nit: IsBinary or IsImage. IsAssembly would require us to peek into image header.